docs/sts-migrate-to-private-bucket.md

```diff
@@ -0,0 +1,428 @@
+export OIDC_BUCKET_NAME=$(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} | awk -F'://' '{print$2}' |awk -F'.' '{print$1}')
+export CLUSTER_REGION=$(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} | awk -F'://' '{print$2}' |awk -F'.' '{print$3}')
+```
+
+- Make sure you can reach (read) the S3 Bucket created with the default name
+
+```bash
+aws s3 ls --region $CLUSTER_REGION s3://$OIDC_BUCKET_NAME
+```
+
+- A clean work directory: a lot of files will be created, make sure you switched to a new work directory to save the files properly (it can be used in the future for rollback)
+
+## Test the existing token <a name="test-token"></a>
+
+This section is to make sure everything is working correctly in your existing environment.
+
+The steps described below will use the credentials provided to the `machine-api` component, trying to assume the role using `aws-cli`.
+
+It's expected that the existing token will be able to authenticate in AWS. Otherwise, you must abort the operations and do not try to run any step described in the next sections
+
+```bash
+## test existing token
+# Get Token path from AWS credentials mounted to pod
+TOKEN_PATH=$(oc get secrets aws-cloud-credentials \
+    -n openshift-machine-api \
+    -o jsonpath='{.data.credentials}' |\
+    base64 -d |\
+    grep "^web_identity_token_file" |\
+    awk '{print$3}')
+
+# Get Controler's pod
+CAPI_POD=$(oc get pods -n openshift-machine-api \
+    -l api=clusterapi \
+    -o jsonpath='{.items[*].metadata.name}')
+
+# Extract tokens from the pod
+TOKEN=$(oc exec -n openshift-machine-api ${CAPI_POD} \
+    -c machine-controller -- cat ${TOKEN_PATH})
+
+echo $TOKEN | awk -F. '{ print $2 }' | base64 -d 2>/dev/null | jq .iss
+
+IAM_ROLE=$(oc get secrets aws-cloud-credentials \
+    -n openshift-machine-api \
+    -o jsonpath='{.data.credentials}' |\
+    base64 -d |\
+    grep "^role_arn" |\
+    awk '{print$3}')
+
+echo $IAM_ROLE
+
+aws sts assume-role-with-web-identity \
+    --role-arn "${IAM_ROLE}" \
+    --role-session-name "my-session" \
+    --web-identity-token "${TOKEN}"
+```
+
+## Setup new OIDC with CloudFront Distribution <a name="setup"></a>
+
+### Create the CloudFront Distribution <a name="setup-cloudfront"></a>
+
+- Create the Origin Access Identity (OAI)
+
+```bash
+export DIR_CCO="./"
+export OIDC_BUCKET_PATH="/pvt"
+
+export OAI_CLOUDFRONT_ID=$(aws cloudfront create-cloud-front-origin-access-identity \
+    --cloud-front-origin-access-identity-config \
+    CallerReference="${OIDC_BUCKET_NAME}",Comment="OAI-${OIDC_BUCKET_NAME}" \
+    | jq -r .CloudFrontOriginAccessIdentity.Id)
+```
+
+- Create the CloudFront Distribution
+```bash
+# Should be updated to master before merging
+#wget https://raw.githubusercontent.com/openshift/cloud-credential-operator/master/docs/sts-oidc-cloudfront.json.tpl
+wget https://raw.githubusercontent.com/mtulio/cloud-credential-operator/doc-sts-updates/docs/sts-oidc-cloudfront.json.tpl
```

Comment:
1) We can use inline json here, to avoid downloading.
2) Do we need to enable the cache policy? The default behavior is not applying any policy. I think setting cache policy to basic cache[A] is fine, while those objects will not be changed, and there is a lot of requests per minute for the same objects - will save costs going to S3.

Possible policy to be used:
![Screenshot from 2023-02-24 00-34-41](https://user-images.githubusercontent.com/3216894/221086017-5ec46da0-7740-4a44-a713-0ba2d9e1f489.png)

If we decide to enable the cache policy, we must consider tracking the ccoctl changes, and fix the original STS document: https://github.com/openshift/cloud-credential-operator/blob/master/docs/sts-private-bucket.md

Reply:
Let's keep the note here, not sure if we need to use cache for now. Customers can enable it if they want.